sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/instrumentation/otel/metrics/OTelMeter.java

```diff
@@ -0,0 +1,127 @@
+ */
+public class OTelMeter implements Meter {
+    private static final ClientLogger LOGGER = new ClientLogger(OTelMeter.class);
+    private static final FallbackInvoker SET_INSTRUMENTATION_VERSION_INVOKER;
+    private static final FallbackInvoker BUILD_INVOKER;
+    private static final FallbackInvoker SET_SCHEMA_URL_INVOKER;
+    private static final FallbackInvoker GET_METER_BUILDER_INVOKER;
+
+    static {
+        ReflectiveInvoker setInstrumentationVersionInvoker = null;
+        ReflectiveInvoker buildInvoker = null;
+        ReflectiveInvoker setSchemaUrlInvoker = null;
+        ReflectiveInvoker getMeterBuilderInvoker = null;
+
+        if (OTelInitializer.isInitialized()) {
+            try {
+                setInstrumentationVersionInvoker = getMethodInvoker(METER_BUILDER_CLASS,
+                    METER_BUILDER_CLASS.getMethod("setInstrumentationVersion", String.class));
+
+                setSchemaUrlInvoker = getMethodInvoker(METER_BUILDER_CLASS,
+                    METER_BUILDER_CLASS.getMethod("setSchemaUrl", String.class));
+
+                buildInvoker = getMethodInvoker(METER_BUILDER_CLASS, METER_BUILDER_CLASS.getMethod("build"));
+
+                getMeterBuilderInvoker = getMethodInvoker(METER_PROVIDER_CLASS,
+                    METER_PROVIDER_CLASS.getMethod("meterBuilder", String.class));
+            } catch (Throwable t) {
+                OTelInitializer.initError(LOGGER, t);
+            }
+        }
+
+        SET_INSTRUMENTATION_VERSION_INVOKER = new FallbackInvoker(setInstrumentationVersionInvoker, LOGGER);
+        SET_SCHEMA_URL_INVOKER = new FallbackInvoker(setSchemaUrlInvoker, LOGGER);
+        BUILD_INVOKER = new FallbackInvoker(buildInvoker, NoopMeter.INSTANCE, LOGGER);
+        GET_METER_BUILDER_INVOKER = new FallbackInvoker(getMeterBuilderInvoker, LOGGER);
+    }
+
+    private final Object otelMeter;
+
+    /**
+     * Creates a new instance of OTelMeter.
+     *
+     * @param otelMeterProvider The OpenTelemetry meter provider.
+     * @param libraryOptions The library options.
+     */
+    public OTelMeter(Object otelMeterProvider, LibraryInstrumentationOptions libraryOptions) {
+        Object meterBuilder = GET_METER_BUILDER_INVOKER.invoke(otelMeterProvider, libraryOptions.getLibraryName());
+        if (meterBuilder != null) {
+            SET_INSTRUMENTATION_VERSION_INVOKER.invoke(meterBuilder, libraryOptions.getLibraryVersion());
+            SET_SCHEMA_URL_INVOKER.invoke(meterBuilder, libraryOptions.getSchemaUrl());
+            this.otelMeter = BUILD_INVOKER.invoke(meterBuilder);
+        } else {
+            this.otelMeter = null;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public DoubleHistogram createDoubleHistogram(String name, String description, String unit) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        Objects.requireNonNull(description, "'description' cannot be null.");
```

Comment:
`unit` can be null?

Reply:
yep, unit is allowed to be null in otel. We can be stricter though